paramiko/proxy.py

```diff
@@ -107,7 +107,21 @@ def recv(self, size):
             raise ProxyCommandFailure(' '.join(self.cmd), e.strerror)
 
     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        kill_after = datetime.datetime.utcnow() + datetime.timedelta(seconds=5)
+        try:
+            self.process.terminate()
+            while datetime.datetime.utcnow() < kill_after:
+                self.process.poll()
+                if self.process.returncode is not None:
```

Comment:
this can be slightly consolidated into

```python
if self.process.poll() is not None:
```

(because `.poll()` returns `returncode`)